### PR TITLE
API returns random movie

### PR DESCRIPTION
**Pull request by Alex-Bellamy**
## User Story
```
As a movie API
When the visitor clicks random
I want to return a random movie from the movie index
```
## Changes proposed in this pull request:
* Adding a feature to random a movie, adding tests for that feature, adding webmock, adding a stub for the api.
## What I have learned working on this feature:
* We have learned how to make a randomising feature, we have learned to use webmock, we have learned to work in a test driven. 
## Packages/Gems added
- webmock, pry-rails, rest-client

### CODE REVIEW

**Review by Cma-B — APPROVED** [review]
it seems fine just remove WIP from the title!

File: spec/requests/api/v0/movies_collection_endpoint_spec.rb [hunk 1]
```diff
@@ -8,12 +8,8 @@
...
     end
 
-  #   it 'is expected to return a collection of movies in an array' do 
-  #     expect(@response_json["results"]).to be_an Array
-  #   end
-
-  #   it 'is expected to return a batman movie' do
-  #     expect(@response_json["results"].first["title"]).to eq 'Batman'
-  #   end
+    it 'is expected to return a movie' do
```
Comment (Odeane) [inline spec/requests/api/v0/movies_collection_endpoint_spec.rb]: I think this should be ......'It is expected to return a movie "title". '

File: app/controllers/api/v1/movies_controller.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Api::V1::MoviesController < ApplicationController
+  def show 
+    random_page = rand(1..500)
+    api_endpoint = "https://api.themoviedb.org/3/discover/movie?api_key=80089a9e6edb5f524156c569cd8a9a69&page=#{random_page}"
+    results = RestClient.get(api_endpoint)
+    movies = JSON.parse(results.body)["results"]
+    random_movie = movies[rand(0..(movies.count))]
```
Comment (emtalen) [inline app/controllers/api/v1/movies_controller.rb]: You need to add what I talked to Viktor about yesterday afternoon, that the movies.count has to be minus one

File: coverage/.resultset.json [hunk 3]
```diff
@@ -1,18 +1,34 @@
 {
   "RSpec": {
     "coverage": {
```
Comment (emtalen) [inline coverage/.resultset.json]: You need to clear the cache as i showed Viktor yesterday, so the coverage is not pushed up everytime

File: spec/requests/api/v1/user_can_randomise_a_movie_spec.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    end
+
+    it 'is expected to get a 200 response' do
+      expect(response).to have_http_status 200
+    end
+
+    it 'is expected to return a movie' do
+      expect(response_json["movie"]).to include("title")
+    end
+
```
Comment (emtalen) [inline spec/requests/api/v1/user_can_randomise_a_movie_spec.rb:14]: When you correct the functionality in the controller you can change the response in the fixture to be one movie and then you can test more specific for that movie.

**Review by emtalen — CHANGES_REQUESTED** [review]
You need to adress the cache of the coverage and the randomising of a movie index, as i talked about with viktor yesterday afternoon

File: config/routes.rb [hunk 5]
```diff
@@ -1,2 +1,7 @@
 Rails.application.routes.draw do
+    namespace :api do
+        namespace :v1 do
+            resources :movies, only: [:show]
+        end
+    end
 end
```
Comment (emtalen) [inline config/routes.rb]: Please mind the indentation, its 4 spaces here and not 2

**Review by emtalen — COMMENTED** [review]
Just a minor fomatting issue now

**Review by emtalen — APPROVED** [review]

**Pull request merged**